mitmproxy/addons/view.py

```diff
@@ -69,30 +70,42 @@ def __call__(self, f):
 
 class OrderRequestStart(_OrderKey):
     def generate(self, f: http.HTTPFlow) -> int:
-        return f.request.timestamp_start or 0
+        if isinstance(f, http.HTTPFlow):
+            return f.request.timestamp_start or 0
+        else:
+            return f.timestamp_start
```

Comment:
We probably have quite a few changes like this, so let's make sure that we do it a bit more systematic:

```python
    def generate(self, f: flow.Flow) -> int:  # note the updated type definition
        if isinstance(f, http.HTTPFlow):
            return f.request.timestamp_start or 0
        elif isinstance(f, tcp.TCPFlow):
            return f.timestamp_start
        else:
            raise NotImplementError(f)
```

This makes it hopefully more straightforward to add additional protocols. For example, your `OrderRequestMethod` would silently fail with "TCP" instead of complaining with an error.